boxsdk/object/base_endpoint.py

```diff
@@ -2,11 +2,13 @@
 
 from __future__ import unicode_literals
 
+from boxsdk.object.base_api_json_object import BaseAPIJSONObject
 
-class BaseEndpoint(object):
+
+class BaseEndpoint(BaseAPIJSONObject):
```

Comment:
Endpoints are more general than objects. For example, see https://github.com/box/box-python-sdk/blob/master/boxsdk/object/search.py#L120

So `BaseEndpoint` should remain unchanged. And `BaseObject` should inherit from both `BaseEndpoint` and `BaseAPIJSONObject`.